### PR TITLE
[SAR-474] - Add submeasures to measure by line chart

### PR DESCRIPTION
**Pull request by anthem123**
# Related Tickets

- [SAR-474](https://jira.amida-tech.com/browse/SAR-474)

# How Things Worked (or Didn't) Before This PR

The measure by line chart would only show the single measure selected. So sub measures were shown and the chart bar wouldn't display.

# How Things Work Now (And How to Test)

Now the measure by line chart works just like the standard chart does. So much so that they are just the same chart with different data. The measure by line shows the selected measure as well as sub measures.

### CODE REVIEW

**Review by eawest2 — APPROVED** [review]
Very line, much submeasure.

**Review by timAmida — APPROVED** [review]

File: src/components/ChartContainer/D3Container.js [hunk 1]
```diff
@@ -86,7 +90,7 @@ function D3Container({ dashboardState, dashboardActions, store }) {
     if (store.currentResults !== undefined) {
       setSelectedMeasures(store.currentResults.map((result) => result.measure));
     }
-  }, [setSelectedMeasures, setCurrentFilters, store.currentResults]);
+  }, [setSelectedMeasures, setCurrentFilters, setByLineSelectedMeasures, store.currentResults]);
```
Comment (jmfadeley) [inline src/components/ChartContainer/D3Container.js]: Does `setByLineSelectedMeasures` need to be added here? It's not being used.

File: src/components/ChartContainer/D3Container.js [hunk 2]
```diff
@@ -192,35 +276,34 @@ function D3Container({ dashboardState, dashboardActions, store }) {
         <Tab label="Measure by Line" className="d3-container__tab-button" />
       </Tabs>
       <TabPanel value={tabValue} index={1}>
-        <Grid container>
+        <Grid container justifyContent="space-evenly" direction="column">
```
Comment (jmfadeley) [inline src/components/ChartContainer/D3Container.js]: Candidate for CSS.

File: src/components/ChartContainer/D3Container.js (hunk 1 above)
Comment (anthem123) [reply src/components/ChartContainer/D3Container.js]: I tested it with it removed and just as you thought, it wasn't doing a thing. I'll remove that.

File: src/components/ChartContainer/D3Container.js [hunk 3]
```diff
@@ -192,35 +276,34 @@ function D3Container({ dashboardState, dashboardActions, store }) {
...
-          currentTimeline={currentTimeline}
+        <MeasureResultsTable
+          currentResults={byLineCurrentResults}
+          handleMeasureChange={handleByLineMeasureChange}
+          selectedMeasures={byLineSelectedMeasures}
+          colorMapping={byLineColorMap}
         />
       </TabPanel>
       <TabPanel value={tabValue} index={0}>
         <Grid container justifyContent="space-evenly" direction="column">
```
Comment (jmfadeley) [inline src/components/ChartContainer/D3Container.js]: Also candidate for CSS, probably using the same class as the one above.

File: src/components/ChartContainer/D3Container.js (hunk 2 above)
Comment (anthem123) [reply src/components/ChartContainer/D3Container.js]: That styling is duplicated too in the other tab panel.

**Review by jmfadeley — APPROVED** [review]
@anthem123 just a point or two, but looks great. If you decide not to do them, please make a 1 point ticket to adjust.

**Pull request merged**